### PR TITLE
Remove support for (undocumented) --compiler argument

### PR DESCRIPTION
**Pull request by astrofrog**
Before this PR, we allowed the compiler returned by ``get_compiler`` to be overridden by the ``--compiler`` command-line argument. This wasn't actually documented and removing support for this allows us to clean up quite a bit of code.

Needs to be rebased once https://github.com/astropy/extension-helpers/pull/34 is merged!

### CODE REVIEW

File: extension_helpers/_setup_helpers.py [hunk 1]
```diff
@@ -7,18 +7,33 @@
...
+
+    Returns
+    -------
+    compiler : str
+        The compiler option specified for the build, build_ext, or build_clib
+        command; or the default compiler for the platform if none was
+        specified.
+
+    """
+    return new_compiler().compiler_type
```
Comment (Cadair) [inline extension_helpers/_setup_helpers.py:37]: If this is the upshot of this change then +:100: because it will make #33 a lot easier.

Comment (astrofrog) [reply extension_helpers/_setup_helpers.py]: I mean at some point we can even deprecate ``get_compiler`` but no rush for that

Comment (Cadair) [reply extension_helpers/_setup_helpers.py]: We would need to provide a way to switch on compiler / platform though as different compilers can use different flags etc.

File: extension_helpers/_utils.py [hunk 2]
```diff
@@ -162,7 +127,7 @@ def import_file(filename, name=None):
 
     if name is None:
         basename = os.path.splitext(filename)[0]
-        name = '_'.join(os.path.relpath(basename).split(os.sep)[1:])
+        name = '_'.join(os.path.abspath(basename).split(os.sep)[1:])
```
Comment (Cadair) [inline extension_helpers/_utils.py:130]: :eyes: hmmm, wat.

**Review by Cadair — APPROVED** [review]
All praise the holy fire of technical debt removal. :fire:

File: extension_helpers/_utils.py (hunk 2 above)
Comment (astrofrog) [reply extension_helpers/_utils.py:130]: It doesn't really matter I think, this is just generating a module name which isn't really important? On Windows using relpath causes issues on Azure...

Comment (Cadair) [reply extension_helpers/_utils.py:130]: Cool

**Pull request merged**